pkg/executables/kubectl.go

```diff
@@ -143,6 +143,31 @@ type capiMachinesResponse struct {
 	Items []clusterv1.Machine
 }
 
+// WaitForEKSAControlPlaneAvailable blocks until the first control plane is available for EKSA cluster.
+func (k *Kubectl) WaitForEKSAControlPlaneAvailable(ctx context.Context, cluster *types.Cluster, timeout string, newClusterName string) error {
+	return k.Wait(ctx, cluster.KubeconfigFile, timeout, "ControlPlaneInitialized", fmt.Sprintf("%s/%s", eksaClusterResourceType, newClusterName), constants.DefaultNamespace)
```

Comment:
I would prefer if instead of relying on kubectl, we just implement it ourselves
That way we can check the observed generation on every try

Same for the other conditions
We can just have a `WaitForClusterCondition(ctx, client, cluster, condition)` method